pkg/datapath/linux/node.go

```diff
@@ -902,8 +899,8 @@ func (n *linuxNodeHandler) nodeUpdate(oldNode, newNode *nodeTypes.Node, firstAdd
 		newKey = newNode.EncryptionKey
 	}
 
-	if n.enableNeighDiscovery {
-		n.insertNeighbor(context.Background(), newNode, false)
+	if n.enableNeighDiscovery && !newNode.IsLocal() {
+		go n.insertNeighbor(context.Background(), newNode, false)
```

Comment:
very tiny nit: I think this `go` statement here makes two assumptions. Both of which seem valid, but maybe we should leave a comment about it here.

1. `newNode` is now accessed concurrently by multiple go routines. All access seem to be reads (I haven't found any writes), so it seems safe.
2. It assumes it is safe to invoke `insertNeighbor` for the same node concurrently. Reading the rest of the commits, this also seems safe, but I'm less confident here.

Reply:
Yep, makes sense, I will add it.